packages/sanity/src/core/bundles/hooks/usePerspective.tsx

```diff
@@ -1,5 +1,9 @@
 import {useRouter} from 'sanity/router'
 
+// TODO: Importing `usePaneRouter` like this results in an error: `Cannot access 'ScrollContainer' before initialization`.
+// import {usePaneRouter} from 'sanity/structure'
```

Comment:
Could it be that the  issue is caused because you cannot import from `structure` into `core`

Reply:
That might be the case, but there doesn't seem to be a linting rule preventing it. Either way, it might be a sign that we need to restructure the context usage here. Maybe `usePerspective` should move 🤔.

Reply:
I think it's fine that `usePerspective` is part of core, do you think otherwise? 

Maybe the `nearestPerspective` needs to be part of the props received received in form builder? 
The form builder is used in structure to render the form, in the case of a "pane perspective" it can be exposed into the form from in https://github.com/sanity-io/sanity/blob/feat/corel-3/multiple-document-versions/packages/sanity/src/structure/panes/document/documentPanel/documentViews/FormView.tsx#L182-L183

Reply:
Resolved through a wider restructure. Thank you for your help!